libraries/botframework-config/src/models/qnaMakerService.ts

```diff
@@ -39,8 +39,14 @@ export class QnaMakerService extends ConnectedService implements IQnAService {
      */
     constructor(source: IQnAService = {} as IQnAService) {
         super(source, ServiceTypes.QnA);
+        
+        if (!source.hostname) {
+            throw TypeError('QnAMakerService requires source parameter to have a hostname.')
+        }
 
-        this.hostname = new URL('/qnamaker', this.hostname).href;
+        if (!this.hostname.endsWith('/qnamaker')) {
```

Comment:
If I'm being picky, "/qnamaker" isn't a host name. 

"/qnamaker" is a path segment. 

In this case, I think the underlying IQnAService is messed up, so we can't change it.